### PR TITLE
Configure nginx proxy to split api/app traffic

### PR DESCRIPTION
**Pull request by wbprice**

### CODE REVIEW

File: reverse-proxy/nginx.conf [hunk 1]
```diff
@@ -0,0 +1,22 @@
+worker_processes 1;
+
+events { worker_connections 1024; }
```
Comment (qwo) [inline reverse-proxy/nginx.conf]: for a setting with only one option you don't need the braces so i would rewrite 
`events  worker_connections 1024;`

File: reverse-proxy/nginx.conf [hunk 2]
```diff
@@ -0,0 +1,22 @@
+worker_processes 1;
```
Comment (qwo) [inline reverse-proxy/nginx.conf]: For this I think its suggested to run it as "auto"
Hard coding 1 sound wrong but I can't find docs to what a proper node configuration would be

Comment (qwo) [reply reverse-proxy/nginx.conf]: see where for 
`worker_processes auto;`
https://gist.github.com/denji/8359866

**Review by BretFisher — CHANGES_REQUESTED** [review]
1. no need to build proxy image for local dev, just use default nginx image and use volumes to bind-mount the conf into it at runtime. 
2. likely no need to edit `/etc/nginx/nginx.conf`, but rather bind-mount over the default site config to make it much easier: `/etc/nginx/conf.d/default.conf`. Look at that file in container to see example (it's just the `server{}` part).

**Comment by wbprice** [comment]
@BretFisher @stanzheng updated

**Comment by qwo** [comment]
LGTM and the request @BretFisher requested.

**Pull request merged**